benchmarks/src/jmh/kotlin/benchmarks/flow/FlowFlattenMergeBenchmark.kt

```diff
@@ -0,0 +1,52 @@
+package benchmarks.flow
+
+import doWork
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.runBlocking
+import org.openjdk.jmh.annotations.*
+import java.util.concurrent.TimeUnit
+
+@Warmup(iterations = 3)
+@Measurement(iterations = 10)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Benchmark)
+@Fork(1)
+open class FlowFlattenMergeBenchmark {
+
+    /**
+     * Number of flows that are merged in this benchmark. Negative number means that number of flows
+     * will be computed as -([flows] * [concurrency]), positive number will be chosen as number of flows.
+     */
+    @Param("-10", "-1", "100", "500")
+    private var flows: Int = 0
+
+    @Param("1", "2", "4") // local machine
+//    @Param("1", "2", "4", "8", "16", "32", "64", "96") // Google Cloud
+    private var concurrency: Int = 0
+
+    private lateinit var flow: Flow<Flow<Int>>
+
+    @Setup
+    fun setup() {
+        val n = if (flows >= 0) flows else -(flows * concurrency)
+        val flowElementsToProcess = ELEMENTS / n
+
+        flow = (1..n).asFlow().map {
+            flow {
+                repeat(flowElementsToProcess) {
+                    doWork(WORK)
```

Comment:
Why do you have your own `doWork` instead of battle-tested `Blackhole.consumeCpu`?
Have you checked that `doWork(constant)` is not eliminated and has a linear growth along with its parameter? Do we have a benchmark for that?

Reply:
It is better to vary additional work in benchmarks. The best way (at least, in theory) is sampling work sizes with the Poisson distribution. However, it is too expensive to use this distribution at run-time, and the geometric distribution is a good trade-off (it is also randomized and is close to the Poisson one).